index.js

```diff
@@ -3778,6 +3778,28 @@
   S.pairs =
   def('pairs', {}, [$.StrMap(a), $.Array($.Pair($.String, a))], pairs);
 
+  //# fromPairs :: Array (Pair String a) -> StrMap a
+  //.
+  //. Returns a string map containing the key–value pairs specified by
+  //. the given array. If a key appears in multiple pairs, the rightmost
+  //. pair takes precedence.
+  //.
+  //. ```javascript
+  //. > S.fromPairs([['a', 1], ['b', 2], ['c', 3]])
+  //. {a: 1, b: 2, c: 3}
+  //.
+  //. > S.fromPairs([['x', 1], ['x', 2]])
+  //. {x: 2}
+  //. ```
+  function fromPairs(pairs) {
+    return Z.reduce(function(strMap, pair) {
+      strMap[pair[0]] = pair[1];
+      return strMap;
+    }, {}, pairs);
+  }
+  S.fromPairs =
+  def('fromPairs', {}, [$.Array($.Pair($.String, a)), $.StrMap(a)], fromPairs);
```

Comment:
Since the function uses `Z.reduce`, maybe this signature could be expanded to:

```hs
Foldable f => f (Pair String a) -> StrMap a
```

Reply:
Good idea! Let's replace `array` with `[Foldable][]` in the description while we're at it. :)